### PR TITLE
Additional jumpy functions

### PR DESCRIPTION
**Pull request by DavidSlayback**
- Control flow:
  - While loop
  - Cond
- Random:
  - randint
  - choice
- Miscellaneous
  - atleast_1/2/3d
  - meshgrid

Following up from #174, these are the additional jumpy functions I wound up implementing for my own brax environments.

### CODE REVIEW

**Comment by jkterry1** [comment]
@erikfrey @cdfreeman-google what's the status of this just because this will have knock on impacts for Gym?

File: brax/jumpy.py [hunk 1]
```diff
@@ -114,6 +114,32 @@ def scan(f: Callable[[Carry, X], Tuple[Carry, Y]],
...
+
+def fori_loop(lower: int, upper: int,
+               body_fun: Callable[[X], X],
+               init_val: X) -> X:
+  """Call body function over range from lower to upper, starting with state"""
+  if _in_jit():
+    return jax.lax.fori_loop(lower, upper, body_fun, init_val)
+  else:
+    val = init_val
+    for i in range(lower, upper):
```
Comment (erikfrey) [inline brax/jumpy.py:138]: nit: some linters will complain about this.  prefer:

```
for _ in range(lower, upper)
```

**Comment by erikfrey** [comment]
This is really nicely done, clean code!  Just one nit and we're good to go.

**Review by erikfrey — CHANGES_REQUESTED** [review]
This is really nicely done, clean code! Just one nit and we're good to go.

**Comment by erikfrey** [comment]
@jkterry1 thanks for the ping - just replied.  We'll merge this after one nit.

**Comment by DavidSlayback** [comment]
Thanks for the kind words about the cleanliness, but I really just copied the original coding style! Made the requested changes

**Comment by erikfrey** [comment]
Test failure but it's unrelated to these changes - merging.

**Pull request merged**